libs/platform/ebpf_epoch.c

```diff
@@ -606,27 +485,32 @@ _ebpf_epoch_update_thread_state(uint32_t cpu_id, uintptr_t thread_id, int64_t cu
 {
     ebpf_result_t return_value;
     ebpf_lock_state_t lock_state;
-    ebpf_epoch_thread_entry_t* thread_state;
-    ebpf_epoch_thread_entry_t local_thread_state = {enter ? current_epoch : 0, !enter ? current_epoch : 0};
-    lock_state = ebpf_lock_lock(&_ebpf_epoch_cpu_table[cpu_id].thread_table_lock);
+    ebpf_epoch_state_t* thread_epoch;
+    ebpf_epoch_state_t local_thread_epoch = {current_epoch, true};
+    bool active_entry_found = false;
+    lock_state = ebpf_lock_lock(&_ebpf_epoch_cpu_table[cpu_id].lock);
     return_value = ebpf_hash_table_find(
-        _ebpf_epoch_cpu_table[cpu_id].thread_table, (uint8_t*)&thread_id, (uint8_t**)&thread_state);
+        _ebpf_epoch_cpu_table[cpu_id].thread_table, (uint8_t*)&thread_id, (uint8_t**)&thread_epoch);
     if (return_value == EBPF_SUCCESS) {
         if (enter) {
-            thread_state->entry_epoch = current_epoch;
+            thread_epoch->epoch = current_epoch;
+            thread_epoch->active = true;
         } else {
-            thread_state->exit_epoch = current_epoch;
+            if (thread_epoch->active) {
+                thread_epoch->active = false;
+                active_entry_found = true;
+            }
```

Comment:
suggest: delete found inactive entries. Run benchmark test to see CPU hit due to this.